Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/CartItemRestController.java

```diff
@@ -28,73 +29,123 @@ public CartItemRestController(CartItemRepository cartItemRepo, ListingRepository
         this.listingRepo = listingRepo;
     }
 
+    @GetMapping("/orderHistory")
+    @ResponseBody()
+    public List<CartItem> getOrderHistoryCartItems(@CurrentUser UserPrincipal userPrincipal) {
+        return cartItemRepo.findAllByUserIdAndPurchaseDateTimeIsNotNullOrderByPurchaseDateTimeDesc(userPrincipal.getId());
+    }
+
+    @PostMapping("/checkout")
```

Comment:
This is posting your cart when you are paying?

Reply:
yes, I see it as a post because the checkout process creates order history